lib/ansible/plugins/connection/ssh.py

```diff
@@ -768,6 +771,44 @@ def _send_initial_data(self, fh, in_data, ssh_process):
 
         display.debug(u'Sent initial data (%d bytes)' % len(in_data))
 
+    def _send_initial_file_data(self, fh, in_fh, ssh_process):
```

Comment:
This is not needed, the 'in_data' is written to the pipe when provided. This just creates another way of doing the same thing.

Reply:
Without this the line right above this `display.debug(u'Sent initial data (%d bytes)' % len(in_data))` won't work because `IOBase` doesn't implement `__len__`.

Additionally, [this line](https://github.com/ansible/ansible/pull/74119/files/2a6817d8a3891b4ce309b56d11dc6bdbcc6310ec#diff-f69a630433cbce2e546a7023edeca91878d27069c0f725c3854d40f7e66b81e0R45)

https://github.com/LoveIsGrief/ansible/blob/ansible-builtin-pipe/lib/ansible/plugins/action/pipe.py#L45

would have to be changed to `in_data=local_proc.stdout.read(),`. I'm quite certain that means reading the entire `stdout` into memory. [`send_initial_data` also calls `to_bytes`](https://github.com/ansible/ansible/pull/74119/files/2a6817d8a3891b4ce309b56d11dc6bdbcc6310ec#diff-38cec806ea1a1ee7c3a286c7865334ecfba7ccc49e21d4e8fb8ec1b17938fda6R759) which will ultimately convert the `IOBase` object to a string.